tools/codeformat.py

```diff
@@ -112,6 +114,10 @@ def fixup_c(filename):
                         # This #-line does not need dedenting.
                         dedent_stack.append(-1)
                 else:
+                    if len(dedent_stack) == 0:
+                        raise IndexError(
```

Comment:
I would be inclined to make a custom `CodeFormatError` since `IndexError` isn't really what the actual problem is.